cloud/gcp/metadata.go

```diff
@@ -0,0 +1,23 @@
+package gcp
+
+import (
+	"cloud.google.com/go/compute/metadata"
+	"github.com/treeverse/lakefs/logging"
+)
+
+type MetadataProvider struct {
+	logger logging.Logger
+}
+
+func NewMetadataProvider(logger logging.Logger) *MetadataProvider {
+	return &MetadataProvider{logger: logger}
+}
+
+func (m *MetadataProvider) GetMetadata() map[string]string {
+	projectID, err := metadata.NumericProjectID()
+	if err != nil {
+		m.logger.Warnf("%v: failed to get Google numeric project ID from instance metadata", err)
+		return nil
+	}
+	return map[string]string{"google_numeric_project_id": projectID}
```

Comment:
another thing that's missing (sorry for not catching this earlier) is one way hashing: we never want to send the actual account id, just to know that different installations come from the same one. I suggest hashing the received id here and in the aws provider as well.

in terms of metadata fields, i would seperate it into 2 fields: account_type ("aws", "gcp", etc) and "account_id" which is the hash. would make it easier downstream to consume.